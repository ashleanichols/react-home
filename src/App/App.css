@import url("https://fonts.googleapis.com/css2?family=Playfair+Display&display=swap");
body {
  width: 100%; height: 100%; top: 0; left: 0;
  font-family: "Times", serif;
}