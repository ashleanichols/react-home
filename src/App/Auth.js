import React from 'react'
import { StoreContext } from '../Store/Store'
import { loginSuccess } from '../Store/actions'
import userService from '../services/user-service'
const Auth = ({ children }) => {
    const { dispatch } = React.useContext(StoreContext)
    React.useEffect(() => {
      userService.home()
        .then((res) => {
          console.log("response ", res)
          res.status === 200
            ? res.json()
            : res.text().then((text) => Promise.reject(text))
        })
        .then((user) => {
          console.log(user, ' user')
          dispatch(loginSuccess(user))
        })
        .catch(() => {
          dispatch(loginSuccess(null))
        })
    }, [])
    return <>{children}</>
  }

export default Auth