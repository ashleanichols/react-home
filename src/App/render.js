import React from "react"
import Main from "./Main/Main"

function render(Cmp, { isLogged, ...otherProps}) {
  return function (props) {
    return (
      <Main>
        <Cmp {...props}  {...otherProps} />
      </Main>
    )
  }
}

export default render