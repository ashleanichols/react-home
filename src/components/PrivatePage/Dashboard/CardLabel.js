import React from "react";
import styles from "../styles/card.module.css";
import { Link } from "react-router-dom";
const CardLabel = (props) => {
  return (
    <div className={styles[props.mainStyle]}>
      <img className={styles[props.classImage]} src={props.image} alt="img" />
      <Link to="/api/products/interior">
        <button className={styles["button"]}>Check Now</button>
      </Link>
      <h1 className={styles[props.text]}>
        {props.title}
        <hr className={styles[props.hr]}></hr>{" "}
        <p className={styles[props.p]}>
          <span className={styles[props.span]}>Checkout our designs</span>
          <img className={styles[props.arrow]} src={props.icon} alt="img" />
        </p>
      </h1>
    </div>
  );
};
export default CardLabel;
