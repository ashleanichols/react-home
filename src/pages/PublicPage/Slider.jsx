import React from 'react'
import styles from './styles/slider.module.css'
import Redesign from './Redesign/Redesign'
import Box from './Box'
const Slider = () => {
  return (
    <main>
      <section>
        <div className={styles.container}>
          <div className={styles.slider}>
            <Box mainStyle="box1-box" title="Interior Design" description="You can seek through the new and modern interior designs. Be
          inspired to change your home interior into a beautiful
          comfortable place." buttonStyle="box1-button" />
            <Box mainStyle="box5-box" title="Smart Home Technology" description="Home Automation is the best thing in 2020. Connecting all of
                  your appliances to your smartphone is an obvious plus – it
                  means that you can run your laundry while you’re at work, or
                  save money on your energy bill by optimizing your heating to
                  only be on when you’re home (and maybe 15 minutes either
                  side)." buttonStyle="box5-button" />
          </div>
        </div>
      </section>
      <Redesign />
    </main>
  )
}
export default Slider