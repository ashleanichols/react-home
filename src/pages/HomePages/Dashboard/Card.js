import React from "react";
import styles from "../../../styles/card.module.css";
import CardLabel from './CardLabel'
const Card = () => {
  return (
    <div className={styles["select-two"]}>
      <div className={styles.content}>
        <div className={styles.card}>
          <p>.</p>
          <CardLabel
            mainStyle="box"
            classImage="intr"
            image="https://cdn.dribbble.com/users/93860/screenshots/9710223/media/dfa76676fcbb139505556fd75b40de69.gif"
            text="interior-text"
            title="Interior Design"
            linkTo="/api/products/interior"
            hr="interior-hr"
            p="interior-p"
            span="interior-span"
            arrow="arrow-image"
            icon={require("../../../static/images/next.png")}
          />
          <CardLabel
            mainStyle="box-second"
            classImage="intr"
            image="https://i.pinimg.com/originals/c3/a6/cb/c3a6cbbc1e1d5fed032ea0a49bb9f545.gif"
            text="smart-text"
            linkTo='/api/products/smart'
            title="Smart Home"
            hr="smart-hr"
            p="smart-p"
            span="smart-span"
            arrow="smart"
            icon={require("../../../static/images/return.png")}
          />
        </div>
      </div>
    </div>
  );
};
export default Card;
