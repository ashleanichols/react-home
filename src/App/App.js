import React from "react"
import "./App.css"
import Menu from "../components/PrivatePage/Menu/Menu"
import Slider from "../components/PublicPage/Slider"
import Dashboard from "../components/PrivatePage/Dashboard/Dashboard"
import { Switch, Redirect, Route } from "react-router-dom"
import SignIn from "../components/Authentication/SignIn/SignInLabel"
import Logout from "../components/Authentication/Logout"
import Register from "../components/Authentication/SignUp"
import render from "./render"
import userService from "../services/user-service"
import Interior from "../components/PrivatePage/InteriorDesign"
import Footer from "../components/Footer"
import Header from "../components/Header"
import NotFound from "../components/Error/404"
import notFoundCmp from "../components/Error/404.2"
import Profile from "../components/Profile"
import AddToPage from "../components/PrivatePage/AddToCard"
import BasketPage from "../components/Basket"
import parseCookeis from "../parseCookies"
import jwt from "jsonwebtoken"
import token from "../jwtCookie"

class App extends React.Component {
  constructor(props) {
    super(props)
    const cookies = parseCookeis()
    const isLogged = !!cookies["auth_cookie"]
    this.state = { isLogged, user: '' }
  }
  login = (history, data) => {
    return userService.login(data).then(async (data) => {
      if (data.ok) {
        this.setState({ isLogged: true })
        history.push("/")
      }
    })
  }
  logout = (history) => {
    userService.logout().then(() => {
      this.setState({ isLogged: false })
      history.push("/")
      return null
    })
  }
  render() {
    const { isLogged } = this.state
    let user = ''
    if (token !== undefined) {
      const decodedObj= jwt.verify(token, "secret123")
      user = decodedObj.username
    }
    return (
      <div className="App">
        {!isLogged ? (
          <Header isLogged={isLogged} />
        ) : (
          <Menu isLogged={isLogged} user={user} />
        )}
        <Switch>
          <Route
            path="/"
            exact
            component={
              isLogged
                ? render(Dashboard, { isLogged })
                : render(Slider, { isLogged })  }  />
          <Route
            path="/signin"
            component={
              !isLogged
                ? render(SignIn, { isLogged, login: this.login })
                : () => <Redirect to="/" />  } />
          <Route
            path="/register"
            render={
              !isLogged
                ? render(Register, { isLogged })
                : () => <Redirect to="/" />   } />
          <Route
            path="/logout"
            render={render(Logout, { isLogged, logout: this.logout })} />
          <Route path="/profile/:userid" component={ Profile } />
          <Route path="/interior" render={render(Interior, { isLogged })} />
          <Route path="/addTo" render={render(AddToPage, { isLogged })} />
          <Route path="/api/user/basket/:id" render={render(BasketPage, { isLogged })} />
          {isLogged ? (
            <Route component={NotFound} />
          ) : (
            <Route component={notFoundCmp} />  )}
        </Switch>
        <Footer />
      </div>
    )
  }
}
export default App