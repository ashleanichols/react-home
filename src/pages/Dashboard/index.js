import React from "react";
import styles from "../../styles/dashboard.module.css";
import Card from './Card'
const Dashboard = () => {
  return (
      <div className={styles.dashboard}>
        <Card />
    </div>
  );
};
export default Dashboard;
