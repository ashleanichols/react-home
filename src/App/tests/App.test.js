import React from "react";
import renderer from "react-test-renderer";
import TestingEnvironment from "../../utils/tests/index";
import App from "../App";
import Dashboard from "../../pages/HomePages/Dashboard";
import Slider from "../../pages/HomePages/PublicPage/Slider";

describe("App", () => {
  it("should run App", () => {
    const tree = renderer
      .create(
        <TestingEnvironment value={{ user: { isLogged: true } }}>
          <App />
        </TestingEnvironment>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should run non-authenticated page", () => {
    const tree = renderer
      .create(
        <TestingEnvironment value={{ user: { isLogged: false } }}>
          <Slider />
        </TestingEnvironment>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render authentictad page", () => {
    const tree = renderer.create(
      <TestingEnvironment
        value={{ user: { isLogged: true } }}
      >
        <Dashboard />
      </TestingEnvironment>
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });
});
