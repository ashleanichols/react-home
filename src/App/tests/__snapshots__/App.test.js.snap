// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render authentictad page 1`] = `
<div
  className="dashboard"
>
  <div
    className="select-two"
  >
    <div
      className="content"
    >
      <div
        className="card"
      >
        <p>
          .
        </p>
        <div
          className="box"
        >
          <img
            alt="img"
            className="intr"
            src="https://cdn.dribbble.com/users/93860/screenshots/9710223/media/dfa76676fcbb139505556fd75b40de69.gif"
          />
          <li>
            <a
              href="/api/products/interior"
              onClick={[Function]}
            >
              <button
                className="button"
                type="button"
              >
                Check now
              </button>
            </a>
          </li>
          <h1
            className="interior-text"
          >
            Interior Design
            <hr
              className="interior-hr"
            />
             
            <p
              className="interior-p"
            >
              <span
                className="interior-span"
              >
                Checkout our designs
              </span>
              <img
                alt="img"
                className="arrow-image"
                src="next.png"
              />
            </p>
          </h1>
        </div>
        <div
          className="box-second"
        >
          <img
            alt="img"
            className="intr"
            src="https://i.pinimg.com/originals/c3/a6/cb/c3a6cbbc1e1d5fed032ea0a49bb9f545.gif"
          />
          <li>
            <a
              href="/api/products/smart"
              onClick={[Function]}
            >
              <button
                className="button"
                type="button"
              >
                Check now
              </button>
            </a>
          </li>
          <h1
            className="smart-text"
          >
            Smart Home
            <hr
              className="smart-hr"
            />
             
            <p
              className="smart-p"
            >
              <span
                className="smart-span"
              >
                Checkout our designs
              </span>
              <img
                alt="img"
                className="smart"
                src="return.png"
              />
            </p>
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`App should run App 1`] = `
<div
  className="loader-center"
>
  <span
    className="fa"
  >
    <img
      alt="loader"
      className="gif"
      src="https://cdn.lowgif.com/full/dc86e54ceca03be4-loading-spinner-animated-gif-83320-mediabin.gif"
    />
  </span>
  <img
    alt="loader"
    src="https://i.gifer.com/ZZ5H.gif"
  />
</div>
`;

exports[`App should run non-authenticated page 1`] = `
<main>
  <section>
    <div
      className="container"
    >
      <div
        className="slider"
      >
        <div
          className="box1-box"
        >
          <div
            className="bg"
          />
          <div
            className="details"
          >
            <h1>
              Interior Design
            </h1>
            <p>
              You can seek through the new and modern interior designs. Be
 inspired to change your home interior into a beautiful
 comfortable place.
            </p>
            <li>
              <a
                href="/api/user/signin"
                onClick={[Function]}
              >
                <button
                  className="box1-button"
                  id="redirect-btn"
                  type="button"
                >
                  Check Now
                </button>
              </a>
            </li>
          </div>
          <div
            className="illustration"
          >
            <div
              className="inner"
            />
          </div>
        </div>
        <div
          className="box5-box"
        >
          <div
            className="bg"
          />
          <div
            className="details"
          >
            <h1>
              Smart Home Technology
            </h1>
            <p>
              Home Automation is the best thing in 2020. Connecting all of
 your appliances to your smartphone is an obvious plus – it
 means that you can run your laundry while you’re at work, or
 save money on your energy bill by optimizing your heating to
 only be on when you’re home (and maybe 15 minutes either
 side).
            </p>
            <li>
              <a
                href="/api/user/signin"
                onClick={[Function]}
              >
                <button
                  className="box5-button"
                  id="redirect-btn"
                  type="button"
                >
                  Check Now
                </button>
              </a>
            </li>
          </div>
          <div
            className="illustration"
          >
            <div
              className="inner"
            />
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    id="why-redesign"
  >
    <h1>
      Why redesign and automate your home?
    </h1>
    <article>
      <h2>
        Flexibility
      </h2>
      <p>
        Being able to integrate these newcomers seamlessly will make your job
 as a homeowner much easier, and allow you to keep upgrading to the
 latest lifestyle technology.
      </p>
    </article>
    <article>
      <h2>
        Beautiful and comfortable
      </h2>
      <p>
        Your home interior design based on your taste is a home with a story of who you are. 
 Make the best possible use of the available space and make it significant.
      </p>
    </article>
    <article>
      <h2>
        Maximizing home security
      </h2>
      <p>
        Home Automation systems can connect motion detectors, surveillance
 cameras, automated door locks, and other security measures
 throughout your home so you can activate them from one mobile device
 before heading to bed.
      </p>
    </article>
  </section>
</main>
`;
